pkg/daemon/ceph/client/filesystem_test.go

```diff
@@ -166,3 +167,195 @@ func TestFilesystemRemove(t *testing.T) {
+		logger.Infof("Command: %s %v", command, args)
+		if args[0] == "fs" {
+			if args[1] == "get" {
+				output, err := json.Marshal(fs)
+				assert.Nil(t, err)
+				return string(output), nil
+			}
+			if args[1] == "rm" {
+				return "", nil
+			}
+		}
+		if args[0] == "mds" {
+			if args[1] == "fail" {
+				return "", nil
+			}
+		}
+		return "", errors.Errorf("unexpected ceph command %q", args)
+	}
+
+	err := FailAllStandbyReplayMDS(context, AdminClusterInfo("mycluster"), fs.MDSMap.FilesystemName)
+	assert.Nil(t, err)
+
+	fs = CephFilesystemDetails{
+		ID: 1,
+		MDSMap: MDSMap{
+			FilesystemName: "myfs1",
+			MetadataPool:   2,
+			Up: map[string]int{
+				"mds_0": 123,
+			},
+			DataPools: []int{3},
+			Info: map[string]MDSInfo{
+				"gid_123": {
+					GID:   123,
+					State: "up:active",
+					Name:  fmt.Sprintf("%s-%s", "myfs1", "a"),
+				},
+				"gid_124": {
+					GID:   124,
+					State: "up:standby",
+					Name:  fmt.Sprintf("%s-%s", "myfs1", "b"),
+				},
+			},
+		},
+	}
+	executor.MockExecuteCommandWithOutputFile = func(command, outputFile string, args ...string) (string, error) {
+		logger.Infof("Command: %s %v", command, args)
+		if args[0] == "fs" {
+			if args[1] == "get" {
+				output, err := json.Marshal(fs)
+				assert.Nil(t, err)
+				return string(output), nil
+			}
+			if args[1] == "rm" {
+				return "", nil
+			}
+		}
+		if args[0] == "mds" {
+			if args[1] == "fail" {
+				return "", errors.Errorf("unexpected execution of mds fail")
+			}
+		}
+		return "", errors.Errorf("unexpected ceph command %q", args)
+	}
+	err = FailAllStandbyReplayMDS(context, AdminClusterInfo("mycluster"), fs.MDSMap.FilesystemName)
+	assert.Nil(t, err)
+
+	fs = CephFilesystemDetails{
+		ID: 1,
+		MDSMap: MDSMap{
+			FilesystemName: "myfs1",
+			MetadataPool:   2,
+			Up: map[string]int{
+				"mds_0": 123,
+			},
+			DataPools: []int{3},
+			Info: map[string]MDSInfo{
+				"gid_123": {
+					GID:   123,
+					State: "up:active",
+					Name:  fmt.Sprintf("%s-%s", "myfs1", "a"),
+				},
+				"gid_124": {
+					GID:   124,
+					State: "up:standby-replay",
+					Name:  fmt.Sprintf("%s-%s", "myfs1", "b"),
+				},
+			},
+		},
+	}
+	executor.MockExecuteCommandWithOutputFile = func(command, outputFile string, args ...string) (string, error) {
+		logger.Infof("Command: %s %v", command, args)
+		if args[0] == "fs" {
+			if args[1] == "get" {
+				output, err := json.Marshal(fs)
+				assert.Nil(t, err)
+				return string(output), nil
+			}
+			if args[1] == "rm" {
+				return "", nil
+			}
+		}
+		if args[0] == "mds" {
+			if args[1] == "fail" {
+				return "", errors.Errorf("expected execution of mds fail")
+			}
+		}
+		return "", errors.Errorf("unexpected ceph command %q", args)
+	}
+	err = FailAllStandbyReplayMDS(context, AdminClusterInfo("mycluster"), fs.MDSMap.FilesystemName)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "expected execution of mds fail")
+}
+
+func TestGetMdsIdByRank(t *testing.T) {
```

Comment:
Please also add tests for all error conditions that can be generated by the function. I.e., failed to get ceph fs, fail to get GID, fail to get ID

Reply:
Did you miss the above comment?